### ISSUE TITLE
Does offline storage expire over time?

### ISSUE THREAD

**Issue by iandunn** [xwp/pwa-wp#193]
If a site registers navigation caching routes so that pages are saved offline, will the browser still purge them from the cache if the device is running low on space (Android), or if the site isn't accessed for several weeks (iOS)?

Is that affected by whether or not the user adds the site to their home screen?

For WordCamp.org's use case, if someone buys a ticket for a camp 3 months before the event takes place, and installs the app so that they'll have the schedule while offline, then I think they'll assume that everything will still be cached even if they're running low on space or haven't accessed the site for awhile.

If they haven't added it to their home screen, then it's probably best to let the files be purged, since they haven't expressed interested in keeping them any longer than a normal site would.

**Comment by pkra** [comment]
For iOS, I believe https://medium.com/@firt/progressive-web-apps-on-ios-are-here-d00430dee3a7 is still valid: iOS might remove files if the PWA is not used for a long time. Android might remove them under storage pressure.

**Comment by westonruter** [comment]
If using using the following navigation caching strategy args as found on the readme, then no, there is no automatic time-based expiration:

```php
add_filter( 'wp_service_worker_navigation_caching_strategy_args', function( $args ) {
	$args['cacheName'] = 'pages';
	$args['plugins']['expiration']['maxEntries'] = 50;
	return $args;
} );
```

Rather, the first cached page would be flushed when when the 51st page is visited. To do time-based cache expiration, instead of `maxEntries` you'd rather provide `maxAgeSeconds`. For more on that see https://developers.google.com/web/tools/workbox/modules/workbox-cache-expiration

Now, this is just how the service worker is configured to handle garbage collection. It's entirely possible, as I understand, for the browser to empty a site's cache if it runs out of storage. See [Using the Cache API in the service worker](https://developers.google.com/web/ilt/pwa/caching-files-with-service-worker#using_the_cache_api_in_the_service_worker):

> You are responsible for implementing how your script (service worker) handles updates to the cache. All updates to items in the cache must be explicitly requested; items will not expire and must be deleted. However, if the amount of cached data exceeds the browser's storage limit, the browser will begin evicting all data associated with an origin, one origin at a time, until the storage amount goes under the limit again. See [Browser storage limits and eviction criteria](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Browser_storage_limits_and_eviction_criteria) for more information. 

Basically, when storage is running  out, the site not visited in the longest time would be the first to get its cache purged. 

I don't believe it matters whether or not the user added the site to their homescreen, although it would make sense for a browser to prioritize emptying the caches of old sites not on the homescreen first.

**Comment by westonruter** [comment]
I've also re-asked the question on Stack Overflow: https://stackoverflow.com/q/57044047/93579
